pkg/monitor/cluster/hiveregistrationstatus.go

```diff
@@ -0,0 +1,50 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/hive"
+)
+
+var clusterDeploymentConditionsExpected = map[hivev1.ClusterDeploymentConditionType]corev1.ConditionStatus{
+	hivev1.ClusterReadyCondition:  corev1.ConditionTrue,
+	hivev1.UnreachableCondition:   corev1.ConditionFalse,
+	hivev1.SyncSetFailedCondition: corev1.ConditionFalse,
+}
+
+func (mon *Monitor) emitHiveRegistrationStatus(ctx context.Context) error {
+	if mon.hiveclientset == nil {
+		// TODO(hive): remove this if once we have Hive everywhere
+		mon.log.Info("skipping: no hive cluster manager")
+		return nil
+	}
+
+	if mon.oc.Properties.HiveProfile.Namespace == "" {
+		return fmt.Errorf("cluster %s not adopted. No namespace in the clusterdocument", mon.oc.Name)
```

Comment:
Not asking to change. Just a tip. It is not necessary to have name in the error message: it will be logged and each log line includes everything we need to identify the cluster:

https://github.com/Azure/ARO-RP/blob/98fc429ecefc51d669cc72dc7cd3963231c18f1c/pkg/util/log/log.go#L156-L161